packages/express-did-auth/src/factories/express-middleware-factory.ts

```diff
@@ -0,0 +1,44 @@
+import { JWTPayload } from 'did-jwt'
+import { ACCESS_TOKEN_COOKIE_NAME, ACCESS_TOKEN_HEADER_NAME, DID_AUTH_SCHEME } from '../constants'
+import { ErrorCodes } from '../errors'
+import { verifyAccessToken } from '../jwt-utils'
+import { TokenValidationConfig, RequestCounter } from '../types'
+
+export default function expressMiddlewareFactory (requestCounter: RequestCounter, config: TokenValidationConfig) {
+  return async function (req, res, next) {
+    try {
+      let jwt: string
+      if (config.useCookies) {
+        jwt = req.cookies[ACCESS_TOKEN_COOKIE_NAME]
+      } else {
+        const header = req.headers[ACCESS_TOKEN_HEADER_NAME] || req.headers[ACCESS_TOKEN_HEADER_NAME.toLowerCase()]
+        if (!header) return res.status(401).send(ErrorCodes.NO_ACCESS_TOKEN)
+
+        const splitted = header.split(' ')
+        if (splitted.length !== 2 || splitted[0] !== DID_AUTH_SCHEME) {
+          return res.status(401).send(ErrorCodes.INVALID_HEADER)
+        }
+
+        jwt = splitted[1]
+      }
+
+      if (!jwt) return res.status(401).send(ErrorCodes.NO_ACCESS_TOKEN)
+
+      const verified = await verifyAccessToken(jwt, config)
+      const payload = verified.payload as JWTPayload
```

Comment:
Are we checking the algorithm is not `None`? Is that part of the dependency validations or are we incurring on a vulnerability?

Reply:
It is checked by the `did-jwt` package. You can find that validation here: https://github.com/decentralized-identity/did-jwt/blob/c91d38cdd06635b418250048e329c509eab1e6d6/src/JWT.ts#L300
The allowed signature types are this ones: https://github.com/decentralized-identity/did-jwt/blob/c91d38cdd06635b418250048e329c509eab1e6d6/src/JWT.ts#L76-L80

The JWTs emitted by the `express-did-auth` are signed with `ES256K`